src/js/components/Chart.vue

```diff
@@ -0,0 +1,288 @@
+          <td data-cell="numeric">
+            {{ formatUSD(inputData.operating_income_loss) }}
+          </td>
+        </tr>
+      </tbody>
+    </table>
+  </div>
+</template>
+
+<script>
+import { formatCurrency } from '@/js/utils';
+
+export default {
+  name: 'Chart',
+  props: {
+    inputData: {
+      type: Object,
+      required: true,
+    },
+  },
+  computed: {
+    period() {
+      const str = this.inputData.period_id.toString();
+      return `${str.substring(0, 4)} Q${str.slice(-1)}`;
+    },
+    period_y1() {
+      const str = this.inputData.period_id.toString();
+      const year = (parseInt(str.substring(0, 4), 10) - 1).toString();
+      return `${year} Q${str.slice(-1)}`;
+    },
+    X0() {
+      return 5;
+    },
+    Y0() {
+      return 5;
+    },
+    WIDTH() {
+      return 400;
+    },
+    HEIGHT() {
+      return 300;
+    },
+    rev_delta() {
+      return this.inputData.total_revenue - this.inputData.total_revenue_y1;
+    },
+    cogs_delta() {
+      return (
+        this.inputData.total_cost_of_revenue -
+        this.inputData.total_cost_of_revenue_y1
+      );
+    },
+    sga_delta() {
+      return (
+        this.inputData.total_operating_expenses -
+        this.inputData.total_operating_expenses_y1
+      );
+    },
+    cogs_pct_0() {
+      return (
+        this.inputData.total_cost_of_revenue_y1 /
+        this.inputData.total_revenue_y1
+      );
+    },
+    sga_pct_0() {
+      return (
+        this.inputData.total_operating_expenses_y1 /
+        this.inputData.total_revenue_y1
+      );
+    },
+    cogs_pct_delta() {
+      return this.cogs_delta / this.rev_delta;
+    },
+    sga_pct_delta() {
+      return this.sga_delta / this.rev_delta;
+    },
+    h_cogs_0() {
+      return this.HEIGHT * this.cogs_pct_0;
+    },
+    h_sga_0() {
+      return this.HEIGHT * this.sga_pct_0;
+    },
+    h_cogs_delta() {
+      return this.HEIGHT * this.cogs_pct_delta;
+    },
+    h_sga_delta() {
+      return this.HEIGHT * this.sga_pct_delta;
+    },
+    w_delta() {
+      return this.inputData.total_revenue / this.inputData.total_revenue_y1 - 1;
+    },
+    expenses_0() {
+      return this.h_cogs_0 + this.h_sga_0;
+    },
+    expenses_delta() {
+      return this.h_cogs_delta + this.h_sga_delta;
+    },
+    scale() {
+      return (
+        this.HEIGHT /
+        Math.max(this.expenses_0, this.expenses_delta, this.HEIGHT)
+      );
+    },
+  },
+  methods: {
+    formatUSD(input) {
+      return formatCurrency(input);
+    },
+  },
+};
+</script>
+
+<style lang="scss" scoped>
+.chart-container {
+  display: grid;
+}
+
+rect {
+  stroke: var(--rect-stroke);
+  stroke-width: 2px;
+}
+
+.income {
+  fill: var(--income);
+}
+
+.sga {
+  fill: var(--sga);
+}
+
+.cogs {
+  fill: var(--cogs);
+}
+
+.corner {
+  fill: white;
+}
+
+[data-table] {
+  border-collapse: collapse;
+  box-shadow: 3px 5px 3px #0000006c;
+  color: var(--text);
+  font-family: inherit;
+  width: max-content;
```

Comment:
I'm thinking we will want the text to wrap as needed, so maybe something like:

```suggestion
  max-width: 30rem; // Magic number
  width: 100%;
```

Reply:
changed